### PR TITLE
feature:support protobuf maven plugin

### PR DESCRIPTION
**Pull request by github-ygy**
### Ⅰ. Describe what this PR did
support protobuf maven plugin

### Ⅱ. Does this pull request fix one issue?

support protobuf maven plugin to generate java code

### CODE REVIEW

**Comment by slievrly** [comment]
@github-ygy Why do you need to move the location of the proto file?

File: pom.xml [hunk 1]
```diff
@@ -279,6 +280,29 @@
...
+            <plugin>
+                <groupId>org.xolstice.maven.plugins</groupId>
+                <artifactId>protobuf-maven-plugin</artifactId>
+                <version>${protobuf-maven-plugin.version}</version>
+                <configuration>
+                    <protoSourceRoot>${project.basedir}/src/main/resources/protobuf/io/seata/protocol/transcation/</protoSourceRoot>
+                    <protocArtifact>
+                        com.google.protobuf:protoc:3.3.0:exe:${os.detected.classifier}
+                    </protocArtifact>
+                    <pluginId>grpc-java</pluginId>
```
Comment (ujjboy) [inline pom.xml]: no need to use the custom plugin

File: pom.xml [hunk 2]
```diff
@@ -279,6 +280,29 @@
...
+                    <pluginArtifact>
+                        io.grpc:protoc-gen-grpc-java:1.4.0:exe:${os.detected.classifier}
+                    </pluginArtifact>
+                </configuration>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>compile</goal>
+                            <goal>compile-custom</goal>
```
Comment (ujjboy) [inline pom.xml]: add no need to execute goal `compile-custom`

File: pom.xml (hunk 1 above)
Comment (github-ygy) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (github-ygy) [reply pom.xml]: done

**Pull request merged**